source/Missions/Services/Agents/Packets/AgentData.cs

```diff
@@ -35,6 +35,24 @@ public AgentData(Agent agent)
             }
         }
 
+        public AgentData(
+            Vec3 position,
+            Vec2 movementDirection,
+            Vec3 lookDirection,
+            Vec2 inputVector,
+            AgentEquipmentData agentEquipment,
+            AgentActionData agentActionData,
+            AgentMountData agentMountData)
+        {
+            Position = position;
+            MovementDirection = movementDirection;
+            LookDirection = lookDirection;
+            InputVector = inputVector;
+            AgentEquipment = agentEquipment;
+            ActionData = agentActionData;
+            MountData = agentMountData;
+        }
+
         public void Apply(Agent agent)
```

Comment:
Remove this class entirely and move the functionality here to the `AgentMovement` class

Reply:
let me know if there is a reason this class should still exist, I could be wrong here

Reply:
remove it, functionality should be added to AgentMovement struct.